code/__HELPERS/unsorted.dm

```diff
@@ -1074,7 +1074,14 @@ Turf and target are seperate in case you want to teleport some distance from a t
 //Gets the turf this atom inhabits
 /proc/get_turf(atom/movable/AM)
 	if(istype(AM))
-		return locate(/turf) in AM.locs
+		var/list/atom/checkedatoms = list() //prevent recursion from badmins being dumbasses
+		var/atom/A = AM

```

Comment:
@MrStonedOne can't you skip one step by changing this to ```var/atom/A = AM.loc```
Given that in this case AM will always be a /atom/movable not a turf, thus allowing us to skip the first iteration of the ```while(!isturf(A))``` loop.